lib/helper/uploadResultHelper.py

```diff
@@ -0,0 +1,343 @@
+
+                    # The symbol representing the job group in
+                    # treeherder.allizom.org
+                    'group_symbol': 'Hasal',
+                    'group_name': 'Hasal Performance Test',
+
+                    # TODO: get the real timing from the test runner
+                    'submit_timestamp': timestamp,
+                    'start_timestamp': timestamp,
+                    'end_timestamp': timestamp,
+
+                    'state': 'completed',
+                    'result': 'success',
+
+                    'machine': 'local-machine',
+                    # TODO: read platform test result
+                    'build_platform': {
+                        'platform': self.platform,
+                        'os_name': self.os_name,
+                        'architecture': self.build_arch
+                    },
+                    'machine_platform': {
+                        'platform': self.platform,
+                        'os_name': self.os_name,
+                        'architecture': self.machine_arch
+                    },
+
+                    'option_collection': {'opt': True},
+
+                    # jobs can belong to different tiers
+                    # setting the tier here will determine which tier the job
+                    # belongs to.  However, if a job is set as Tier of 1, but
+                    # belongs to the Tier 2 profile on the server, it will still
+                    # be saved as Tier 2.
+                    'tier': 1,
+
+                    # the ``name`` of the log can be the default of "buildbot_text"
+                    # however, you can use a custom name.  See below.
+                    # TODO: point this to the log when we have them uploaded
+                    'log_references': [
+                        {
+                            'url': 'TBD',
+                            'name': 'test log'
+                        }
+                    ],
+                    # The artifact can contain any kind of structured data
+                    # associated with a test.
+                    'artifacts': [
+                        {
+                            'type': 'json',
+                            'name': 'performance_data',
+                            'job_guid': job_guid,
+                            'blob': perf_data
+                        },
+                        {
+                            'type': 'json',
+                            'name': 'Job Info',
+                            'job_guid': job_guid,
+                            'blob': {
+                                'job_details': job_details
+                            }
+                        }
+                    ],
+                    # List of job guids that were coalesced to this job
+                    'coalesced': []
+                }
+            }
+        ]
+
+    def create_job_collection(self, dataset):
+        tjc = TreeherderJobCollection()
```

Comment:
maybe adding a comment to usage page `https://github.com/mozilla/treeherder/blob/master/docs/submitting_data.rst`